rust_src/src/fonts.rs

```diff
@@ -97,6 +103,69 @@ impl FontExtraType {
     }
 }
 
+pub type LispFontObjectRef = ExternalPtr<Lisp_Font_Object>;
+
+impl LispFontObjectRef {
+    pub fn add_log(self, action: &str, result: LispObject) {
+        let c_str = CString::new(action).unwrap();
+        unsafe { font_add_log(c_str.as_ptr(), self.into(), result) }
+    }
+
+    pub fn close(mut self, mut _frame: LispFrameRef) {
+        if data::aref(self.into(), FONT_TYPE_INDEX.into()).is_nil() {
+            // Already closed
+            return;
+        }
+        self.add_log("close", LispObject::from(false));
```

Comment:
In the C code this is a macro call.

```
#define FONT_ADD_LOG(ACTION, ARG, RESULT)       \
  do {                                          \
    if (! EQ (Vfont_log, Qt))                   \
      font_add_log ((ACTION), (ARG), (RESULT)); \
  } while (false)
```

Which means we either a) need a macro or b) need the `add_log` method to check the value before doing anything.

Reply:
I think checking the value is the best solution. However, something that strikes me as odd is that `font_add_log` itself already does the exact same check in the form of
```c
if (EQ (Vfont_log, Qt))
  return;
```
so implementing the check the macro does as well is doing just doing the check twice, which seems kind of pointless.

Reply:
Indeed. I missed that.